core/src/test/java/org/elasticsearch/discovery/DiscoveryWithServiceDisruptionsTests.java

```diff
@@ -1080,4 +1193,33 @@ public boolean apply(Object input) {
             }, 30, TimeUnit.SECONDS));
         }
     }
+
+    private String startDataNode(String color) {
+        return startDataNode(color, createTempDir().toString());
+    }
+
+    private String startDataNode(String color, String newDataPath) {
+        Settings.Builder settingsBuilder = Settings.builder()
+                .put(nodeSettings(0))
+                .put("node.data", true)
+                .put("node.master", false)
+                .put("node.color", color)
+                .put("path.data", newDataPath);
+
+        return internalCluster().startNode(settingsBuilder.build());
+    }
+
+    private String startMasterNode() {
```

Comment:
can we move this to InternalTestCluster ? something like:

```
    public synchronized String startMaserNode(Settings settings) {
        Settings settings1 = Settings.builder().put(settings).put("node.master", true).put("node.data", false).build();
        return startNode(settings1, Version.CURRENT);
    }
```

also add the async variants etc.